docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,418 @@
+
+#### eslintrc Example
+
+```javascript
+// .eslintrc
+
+module.exports = {
+    env: {
+        browser: true,
+    },
+    globals: {
+        myCustomGlobal: "readonly",
+    },
+    parserOptions: {
+        ecmaVersion: 2022,
+        sourceType: "module"
+    }
+    // ...other config
+}
+```
+
+#### Flat Config Example
+
+```javascript
+// eslint.config.js
+
+import globals from "globals";
+
+export default [
+    {
+        languageOptions: {
+            globals: {
+                ...globals.browser,
+                myCustomGlobal: "readonly"
+            },
+            parserOptions: {
+                ecmaVersion: 2022,
+                sourceType: "module"
+            }
+        }
+        // ...other config
+    }
+];
+```
+
+### Predefined Configs
+
+In eslintrc files, use the `extends` property to use predefined configs. ESLint comes with two predefined configs that you can access as strings:
+
+* `"eslint:recommended"`: the rules recommended by ESLint
+* `"eslint:all"`: all rules shipped with ESLint
+
+You can also use the `extends` property to extend a custom config. Custom configs can either be paths to local config files or npm package names.
+
+In flat config files, predefined configs are imported from separate modules into flat confg files. The `recommended` and `all` rules configs are located in the [`@eslint/js`](https://www.npmjs.com/package/@eslint/js) package. You must import this package to use these configs:
+
+```shell
+npm install @eslint/js --save-dev
+```
+
+You can add each of these configs to the exported array or expose specific rules from them. You must import the modules for local config files and npm package configs with flat confg.
+
+#### eslintrc Examples
+
+Use built-in config:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: "eslint:recommended",
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+Use built-in config, local custom config, and custom config from an npm package:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: ["eslint:recommended", "./custom-config.js", "eslint-config-my-config"],
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+#### Flat Config Examples
+
+Use recommended config:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customConfig from "./custom-config.js";
+import myConfig from "eslint-config-my-config";
+
+export default [
+    js.configs.recommended,
+    customConfig,
+    myConfig,
+    {
+        rules: {
+            semi: ["warn", "always"]
+        },
+        // ...other config
+    }
+];
+```
+
+Note that because you are just importing JavaScript modules, you can mutate the config objects before exporting them. For example, you might want to have a certain config object only apply to your test files:
```

Comment:
```suggestion
Note that because you are just importing JavaScript modules, you can mutate the config objects before ESLint uses them. For example, you might want to have a certain config object only apply to your test files:
```